src/content/content_manager.cc

```diff
@@ -659,9 +663,8 @@ void ContentManager::_rescanDirectory(std::shared_ptr<AutoscanDirectory>& adir,
     time_t last_modified_new_max = last_modified_current_max;
     adir->setCurrentLMT(location, 0);
 
-    struct dirent* dent;
-    while ((dent = readdir(dir)) != nullptr) {
-        char* name = dent->d_name;
+    for (const auto& dirEnt : fs::directory_iterator(location, fs::directory_options::skip_permission_denied, ec)) {
```

Comment:
dirEnt is redefined here, I wonder if we can give it a more clear name? or change the other one?